cl/scrapers/management/commands/merge_opinion_versions.py

```diff
@@ -0,0 +1,688 @@
+        found on Opinion.download_url. For example, for `nev` opinions:
+        'http://caseinfo.nvsupreme'
+    :param limit: max number of groups to correct
+    :return None
+    """
+    if url_start:
+        query = get_query_from_url(url_start, "startswith")
+    else:
+        query = Q()
+
+    qs = (
+        Opinion.objects.filter(query)
+        .exclude(Q(download_url="") | Q(download_url__isnull=True))
+        .values("download_url")
+        .annotate(
+            number_of_rows=Count("download_url"),
+            # compute the number of  distinct hashes to prevent colliding with
+            # actual duplicates, which are not versions
+            number_of_hashes=Count("sha1", distinct=True),
+        )
+        .order_by()
+        .filter(number_of_rows__gte=2, number_of_hashes__gte=2)
+    )
+
+    # The groups queryset will look like
+    # {'download_url': 'https://caseinfo.nvsupremecourt...',
+    # 'number_of_rows': 2, 'number_of_hashes': 2}
+    seen_urls = set()
+    stats = defaultdict(lambda: 0)
+    for group in qs:
+        if group["download_url"].replace("https", "http") in seen_urls:
+            continue
+
+        seen_urls.add(group["download_url"].replace("https", "http"))
+
+        if limit and len(seen_urls) > limit:
+            break
+
+        logger.info("Processing group %s", group)
+
+        query = get_query_from_url(group["download_url"], "exact")
+        # keep the latest opinion as the main version
+        # exclude opinions that already have a main_version. If that main
+        # version is a version of the current document, they will be updated
+        # transitively
+        main, *versions = (
+            Opinion.objects.filter(query)
+            .exclude(main_version__isnull=False)
+            .select_related("cluster", "cluster__docket")
+            .order_by("-date_created")
+        )
+        merge_versions_by_text_similarity(main, versions, stats)
+        stats["seen_urls"] = len(seen_urls)
+
+    logger.info(stats)
+
+
+def comparable_dockets(docket: Docket, version_docket: Docket) -> bool:
+    """
+    Make sure that the dockets have at least the same court_id and docket number
+
+    :param docket: the main docket
+    :param version_docket: the version docket
+    :return: True if dockets have the same court_id and docket number
+    """
+    # the same docket
+    if docket.id == version_docket.id:
+        return True
+
+    log_template = "Different '%s' for docket %s and %s"
+
+    if docket.court_id != version_docket.court_id:
+        logger.error(log_template, "court_id", docket.id, version_docket.id)
+        return False
+
+    if docket.docket_number != version_docket.docket_number:
+        logger.error(
+            log_template, "docket_number", docket.id, version_docket.id
+        )
+        return False
+
+    return True
+
+
+def merge_versions_by_text_similarity(
+    main: Opinion, versions: list[Opinion], stats: dict
```

Comment:
I think typing should be QuerySet[Opinion] instead of list[Opinion]

maybe rename main param to main_opinion param to avoid: "if not version.versions.exists():" I think it looks a little weird